static/js/publisher/submitEnabler.js

```diff
@@ -0,0 +1,58 @@
+import shallowDiff from "../libs/shallowDiff";
+
+function submitEnabler(formSelector, buttonSelectors) {
+  if (!formSelector) {
+    throw new TypeError("`formSelector` argument is required");
+  }
+
+  if (!buttonSelectors || buttonSelectors.length === 0) {
+    throw new TypeError("At least one `buttonSelectors` must be defined");
+  }
+
+  const formEl = document.querySelector(formSelector);
+
+  if (!formEl) {
+    throw new Error(`${formSelector} is not a valid element`);
+  }
+
+  const buttonEls = buttonSelectors.map(selector =>
+    document.querySelector(selector)
+  );
+
+  const initialState = new FormData(formEl);
+
+  const initialStateJson = {};
+
+  for (const [key, value] of initialState.entries()) {
+    initialStateJson[key] = value;
+  }
+
+  buttonEls.forEach(button => {
+    if (button) {
+      button.setAttribute("disabled", "disabled");
+      button.classList.add("is--disabled");
+    }
+  });
+
+  formEl.addEventListener("change", () => {
+    const newState = new FormData(formEl);
+    const newStateJson = {};
+
+    for (const [key, value] of newState.entries()) {
+      newStateJson[key] = value;
+    }
+
+    const diff = shallowDiff(initialStateJson, newStateJson);
```

Comment:
Seems like most of the `shallowDiff` logic of marking fields as added/changed/removed, giving previous and current value is not really needed if all we want is a bool check?
Or is it used elsewhere?

Even on listing page where we have a similar check (a possible reuse case) we only list changed/added fields and their values - I'm not sure if we need all of the details `shallowDiff` provides?

Reply:
How do you feel about adding a flag to get either the complete output (or something closer to the listing page diff check), or a boolean?

I could add an issue to update the listing page state stuff to use this.

Reply:
I don't mind - I just felt it does much more that we need it to. Adding even more conditional logic to it seems even more unnecessary ;)

I'm fine with leaving it as it is not to add more work. I just feel that it means we are having a code to maintain that does something we don't need.